### ISSUE TITLE
case/esac with number ranges does not throw SC2221 "pattern always overrides"

### ISSUE THREAD

**Issue by Pitterling** [koalaman/shellcheck#2257]
#### For bugs
- Rule Id (SC2221): 
- My shellcheck version (`0.7.2`): 
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env bash

VERSION='7.3'

case "${VERSION}" in

6.[0-9] | 7.[0-3] ) : ;;

7.2 )   : ;;

6.* )   : ;;

7.* )   : ;;

*):     ;;

esac
```

#### Here's what shellcheck currently says:
No issues detected!

#### Here's what I wanted or expected to see:
Something similar to SC2221 
Line 9 (7.2) is hidden by range 7.[0-3}
